julia/core.py

```diff
@@ -420,16 +420,21 @@ def check_exception(self, src=None):
             # self._debug("No Exception")
             self.api.jl_exception_clear()
             return
-        self._debug("Retrieving exception infos...")
-        stderr = self.api.jl_stderr_obj()
-        self._debug("libjulia stderr = " + str(stderr))
-        self.api.jl_call2(self.api.show, stderr, exoc)
-        self._debug("show called ...")
+        if self.is_debugging:
+            self._debug("Retrieving exception infos...")
+            stderr = self.api.jl_stderr_obj()
+            self._debug("libjulia stderr = " + str(stderr))
+            self.api.jl_call2(self.api.show, stderr, exoc)
+            self._debug("show called ...")
         # self.api.jl_printf(self.api.jl_stderr_stream(), "\n");
 
-        exception_type = self.api.jl_typeof_str(exoc).decode('utf-8')
+        res = self.api.jl_call2(void_p(self.api.convert), void_p(self.api.PyObject), void_p(exoc))
```

Comment:
Probably should be `self.api.jl_call1(void_p(self.api.PyObject), void_p(exoc))` (the `convert` method may go away in a future PyCall version, since this is really a constructor call)

Reply:
I'm a little worried about the interaction of this raw `jl_value_t*` manipulation with [garbage collection in Julia](https://docs.julialang.org/en/release-0.6/manual/embedding/#Memory-Management-1), but I guess that's something that needs to be fixed more generally.

Reply:
Reading git blame, it looks like it is changed from `self.api.jl_call1(void_p(self.api.PyObject), void_p(ans))` to `self.api.jl_call2(void_p(self.api.convert), void_p(self.api.PyObject), void_p(ans))` in https://github.com/JuliaPy/pyjulia/pull/54/commits/0912ac46f5e190b8ae6a64e675fa991b07daeb3c#diff-2f4f81856f2b92c6a4bb1e5bba2cf28aR380 by @Keno (PR #54). Not clear why from git message.

Reply:
Maybe `jl_call1` only works for actual `Function` objects these days, and we'll have to dig to find out how to call the constructor.   Let's continue to use `convert` for now, I guess.